### PR TITLE
dnn: +yolov4

### PR DESCRIPTION
**Pull request by alalek**
**Main PR**: https://github.com/opencv/opencv/pull/17392

### CODE REVIEW

File: testdata/dnn/yolov4.cfg.readme [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[Origin](https://github.com/AlexeyAB/darknet/blob/c0d6b81a78e204c04c9bf4277974e0dadad0c4e2/cfg/yolov4.cfg)
+[License](https://github.com/AlexeyAB/darknet/blob/c0d6b81a78e204c04c9bf4277974e0dadad0c4e2/LICENSE)
```
Comment (dkurt) [inline testdata/dnn/yolov4.cfg.readme:2]: Can we add it to `yolov4.cfg` as comments?
```
# Origin: https://github.com/AlexeyAB/darknet/blob/c0d6b81a78e204c04c9bf4277974e0dadad0c4e2/cfg/yolov4.cfg
# License: https://github.com/AlexeyAB/darknet/blob/c0d6b81a78e204c04c9bf4277974e0dadad0c4e2/LICENSE
```

Comment (alalek) [reply testdata/dnn/yolov4.cfg.readme:2]: I prefer to keep external files without extra modifications.

Comment (dkurt) [reply testdata/dnn/yolov4.cfg.readme:2]: Maybe we can just download it as weights?

Comment (alalek) [reply testdata/dnn/yolov4.cfg.readme:2]: It is better to keep it `opencv_extra` (like others):
- there is no version control on share for downloaded files (they should not conflict at any time)
- configuration file is small and text based
- we may want to add test for importer without weights in the future (without requirement of huge model weights)

**Pull request merged**